pkg/kgo/hooks.go

```diff
@@ -374,6 +374,10 @@ type HookFetchRecordBuffered interface {
 	OnFetchRecordBuffered(*Record)
 }
 
+type HookFetchRecordsBuffered interface {
+	OnFetchRecordsBuffered(topic string, partition int32, records []*Record)
```

Comment:
@twmb Only added this for a talking point. Not happy with this.

We need a way to  inject a root context into all of the messages to share the same root context. We then have a issue getting this context so that we can end the parent span. Currently  OnFetchRecordUnbuffered ends the span but not usable for the batch receive implementation.  


We need someway of storing the context which has been created for the parent span.

Reply:
I was taking a look at this PR again but I don't have an idea for how to solve the unbuffering batch cancel aspect. Do you have an idea for it or is this PR infeasible?

Reply:
Yeah I still don't know how to resolve this. May just be better for now to not have the batch trace and just have the processing trace, to avoid the confusing trace output that happens atm. Will open a fresh PR this week which address that and will close this at time so we can think more about it